### PR TITLE
Added language selection icons

### PR DESCRIPTION
**Pull request by rzats**
### Contains

Adds tiny flag icons to Terasology's language settings menu - similar to (and suggested by a comment in) [TerasologyLauncher#362](https://github.com/MovingBlocks/TerasologyLauncher/pull/362). Technically implements [TerasologyLauncher#60](https://github.com/MovingBlocks/TerasologyLauncher/issues/60).

UI screenshot

A fallback icon (empty 16x11 .png) is displayed if an icon for a locale is missing, and the icon+text renderer is generalized enough to be used for any other purpose (including the NUI editor!)

### How to test

Launch the game, go to _Settings → Player → Language_ and play around with the dropdown.

### Outstanding before merging

No outstanding issues.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/553/
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/rendering/nui/itemRendering/StringTextIconRenderer.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            canvas.drawText(text, Rect2i.createFromMinAndSize(iconWidth, 0, canvas.getRegion().width() - iconWidth, canvas.getRegion().height()));
+        } else {
+            int width = canvas.size().x - iconWidth;
+            Font font = canvas.getCurrentStyle().getFont();
+            if (font.getWidth(text) <= width) {
+                canvas.drawText(text);
+            } else {
+                String shortText = "...";
+                StringBuilder sb = new StringBuilder(text);
+                while (sb.length() > 0) {
```
Comment (msteiger) [inline engine/src/main/java/org/terasology/rendering/nui/itemRendering/StringTextIconRenderer.java]: I believe that similar code exists somewhere already, but that's ok I guess.

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/settings/LocaleRenderer.java [hunk 2]
```diff
@@ -42,5 +45,13 @@ public String getString(Locale locale) {
...
     }
 
-
+    @Override
+    public Texture getTexture(Locale value) {
+        Optional<Texture> texture = Assets.getTexture(String.format("engine:flag_%s", value.getLanguage()));
+        if (texture.isPresent()) {
+            return texture.get();
+        } else {
+            return Assets.getTexture("engine:flag_fallback").get();
```
Comment (msteiger) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/settings/LocaleRenderer.java]: I assume you're doing this to keep the alignment with the other entries?

**Comment by msteiger** [comment]
Great stuff! Code looks good to me. Haven't tested the vertical shrinking / horizontal text truncate / line wrapping.

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/settings/LocaleRenderer.java (hunk 2 above)
Comment (rzats) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/settings/LocaleRenderer.java]: Yep - it's also a more obvious visual indicator that the icon's missing.

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/554/
Hooray Jenkins reported success with all tests good!

**Comment by rzats** [comment]
@msteiger made a small visual test for these:

```java
// in LocaleRenderer.java

@Override
public String getString(Locale locale) {
    String nat = translationSystem.translate("${engine:menu#this-language-native}", locale);
    String eng = translationSystem.translate("${engine:menu#this-language-English}", locale);
    return String.format("%s (%s) Lorem ipsum dolor sit amet, consectetur adipiscing elit", nat, eng);
}

@Override
public Texture getTexture(Locale value) {
    return Assets.getTexture("engine:lavaStill").get(); // lavaStill.png: https://git.io/vVnIM
}
```
test image

Also, pushed another update that made the text wrapping activate by default for the locale setting.

**Comment by Cervator** [comment]
Thanks @rzats ! Hooray for tiny flag icons :-)

I did notice one bug snuck in with the second commit, changing true to false on the word wrap default made the flags and text overlap (huh?) - since it was such a minor change I added it in while merging (needed an excuse to also throw in an update to the credits!) and tested it out as fixed.

If that's a symptom of some other issue feel free to tweak more or wait till the next time some related code needs changing. Wanted you to be aware of it :-)

**Comment by rzats** [comment]
@Cervator - thanks so much for the report! Fixed this in #2272.

**Pull request merged**